app/src/main/java/com/github/se/polyfit/data/remote/firebase/MealFirebaseRepository.kt

```diff
@@ -0,0 +1,77 @@
+package com.github.se.polyfit.data.remote.firebase
+
+import android.util.Log
+import com.github.se.polyfit.model.meal.Meal
+import com.google.android.gms.tasks.Task
+import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.FirebaseFirestore
+
+class MealFirebaseRepository(
+    private val userId: String,
+    private val db: FirebaseFirestore = FirebaseFirestore.getInstance()
+) {
+
+  private val mealCollection = db.collection("users").document(userId).collection("meals")
+
+  fun storeMeal(meal: Meal): Task<DocumentReference> {
+    val mealData = Meal.serialize(meal)
+    return if (meal.firebasaeId.isNotEmpty()) {
+      mealCollection.document(meal.firebasaeId).set(mealData).continueWith { task ->
+        if (task.isSuccessful) {
+          mealCollection.document(meal.firebasaeId)
+        } else {
+          throw task.exception!!
+        }
+      }
+    } else {
+      mealCollection.add(mealData).continueWith { task ->
+        if (task.isSuccessful) {
+          meal.firebasaeId = task.result!!.id
+          task.result
+        } else {
+          throw task.exception!!
+        }
+      }
+    }
+  }
+
+  // maybe ok for now, but in the future will need to find a better way to
+  // distinguish between meals
+  fun getMeal(firebaseID: String): Task<Meal> {
+
+    return getAllMeals().continueWith { task ->
```

Comment:
maybe can use something like this to fetch instead of loading all to the client (not 100% sure on the correct syntax but general idea)
```
    return mealCollection
        .document(firebaseID)
        .get()
```